lib/RefBackend/RefBackend.cpp

```diff
@@ -219,77 +222,89 @@ mlir::torch::RefBackend::createMungeCallingConventionsPass() {
-  Value currentSeed = b.create<memref::LoadOp>(loc, globalVar);
-
-  // The value of multiplier and incrementStep are referenced from
-  // https://en.wikipedia.org/wiki/Linear_congruential_generator for 2^64.
-  Value multiplier = b.create<arith::ConstantOp>(
-      loc, b.getI64IntegerAttr(6364136223846793005));
-  Value incrementStep = b.create<arith::ConstantOp>(
-      loc, b.getI64IntegerAttr(1442695040888963407));
-  // temp = multiplier * currentSeed + incrementStep
-  Value mul = b.create<arith::MulIOp>(loc, currentSeed, multiplier);
-  Value temp = b.create<arith::AddIOp>(loc, mul, incrementStep);
-  // temp mod 64 = temp & 63
-  Value cst127 = b.create<arith::ConstantOp>(loc, b.getI64IntegerAttr(127));
-  Value nextSeed = b.create<arith::AndIOp>(loc, temp, cst127);
-  b.create<memref::StoreOp>(loc, nextSeed, globalVar);
-  return nextSeed;
+static LogicalResult bufferizeMLProgramGlobaLoadOp(ml_program::GlobalLoadOp globalLoadOp,
+                                          OpBuilder &b,
+                                          SmallVector<Operation *> &toErase) {
+  RankedTensorType tensorType = globalLoadOp.getType().cast<RankedTensorType>();
+  MemRefType memrefType =
+      MemRefType::get(tensorType.getShape(), tensorType.getElementType());
+  b.setInsertionPoint(globalLoadOp);
+  Value memref = b.create<memref::GetGlobalOp>(
+      globalLoadOp.getLoc(), memrefType,
+      globalLoadOp.getGlobalAttr().getLeafReference());
+  for (auto user : globalLoadOp->getResult(0).getUsers()) {
+    if (auto extractOp = dyn_cast<tensor::ExtractOp>(user)) {
+      b.setInsertionPoint(extractOp);
+      Value scalar = b.create<memref::LoadOp>(extractOp->getLoc(), memref,
+                                              extractOp.getIndices())
+                         ->getResult(0);
+      extractOp->getResult(0).replaceAllUsesWith(scalar);
+      toErase.push_back(extractOp);
+    } else if (auto insertOp = dyn_cast<tensor::InsertOp>(user)) {
+      b.setInsertionPoint(insertOp);
+      b.create<memref::StoreOp>(insertOp->getLoc(), insertOp.getScalar(),
+                                memref);
+      toErase.push_back(insertOp);
+    }
+  }
+  return success();
 }
 
-// The global seed is stored into a memref<i64> global variable as the only
-// element.
 namespace {
-class InsertRngGlobals : public InsertRngGlobalsBase<InsertRngGlobals> {
+/// Converts MLProgram operations that work on tensor-type operands or results
+/// to work on buffers.
+class MLProgramBufferize : public MLProgramBufferizeBase<MLProgramBufferize> {
   void runOnOperation() override {
     auto module = getOperation();
     OpBuilder b(module.getBodyRegion());
-    createGlobalVariableForSeed(b, module);
     SmallVector<Operation *> toErase;
-    module.walk([&](TorchConversion::GetNextSeedOp op) {
-      b.setInsertionPoint(op);
-      Value seed = lowerGetNextSeed(b, op.getLoc());
-      op.replaceAllUsesWith(seed);
+    module.walk([&](ml_program::GlobalOp op) {
```

Comment:
restrict this to globals with static shape, so that we can verify that the stored memref is the same shape as the global. (and add a test where we reject a dynamically-shaped ml_program.global)